### PR TITLE
Implement a `lock` template in `locks` module.

### PR DESCRIPTION
**Pull request by dom96**

### CODE REVIEW

**Comment by yglukhov** [comment]
s/lock/withLock?

**Comment by dom96** [comment]
Sure, I don't really mind what name we use as long as we have something which abstracts the ugly {.locks.} pragma away for us.

File: lib/core/locks.nim [hunk 1]
```diff
@@ -54,3 +54,11 @@ proc wait*(cond: var Cond, lock: var Lock) {.inline.} =
 proc signal*(cond: var Cond) {.inline.} =
   ## sends a signal to the condition variable `cond`.
   signalSysCond(cond)
+
+template lock*(a: Lock, body: stmt) =
```
Comment (Araq) [inline lib/core/locks.nim]: use 'body: untyped' instead.

**Comment by Araq** [comment]
I agree ``withLock`` is a slightly better name.

**Pull request merged**